framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -552,7 +569,12 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 	}
 
 	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	private async _cleanup() {}
+	private _cleanup() {
+		this._chainConnectorState.crossChainMessages =
+			this._chainConnectorState.crossChainMessages.filter(
```

Comment:
I think filter returns the filtered array so you need to update the original array